server/src/index.ts

```diff
@@ -1,26 +1,30 @@
 import express from "express";
 import bodyParser from "body-parser";
 import db from "./database/db";
+import expressPino from "express-pino-logger";
+import parentLogger from "../utils/logger";
 import scraper from "../utils/scraper";
 
+const log = parentLogger.child({ module: "express" });
```

Comment:
I've never used pino before but how do we know what to pass into the `.child()` function?

Reply:
the cool thing about pino is that the output is just JSON logs, so you can pass really anything into it, and that becomes another piece of information presented in the logs. if i passed `{ module: "express", field: "one" }` instead to `child()`, the output would look something like 

```bash
[1603830499087] INFO     (15544 on DESKTOP-NROP0OM): Server running on port 5000
    module: "express"
    field: "one"

# or in unprettified JSON:
{"level":30,"time":1603830663250,"pid":9280,"hostname":"DESKTOP-NROP0OM","module":"express","field":"one","msg":"Server running on port 5000"}
```

I just passed in the module name to get an idea of what child is posting what

Reply:
Oh I just read the docs so basically this field is always attached to the log. Should we decide on a convention to follow when passing these fields? Personally, I feel that at least for this file, it'd be helpful to know the module is the source index file or something. Any thoughts?

Reply:
hmm that's a good point, maybe we should just name the field `filename` or something instead?

what do you think @johnagl

Reply:
@jacksonyuanjx you can read more about the ideas here in my comment: https://github.com/ubclaunchpad/life-at-ubc/issues/18#issuecomment-717222125

I would strongly recommend against `filename` instead of `module` - having to use `filename` likely indicates your module encapsulates too many features, and the distinction between the files should be turned into separate modules. In logging, you focus is narrowing down where something occurred in your application, and a filename can be very misleading, since frequently that's not where the actual problem is, but just where it happened to be caught and logged (for example, in Rocket our logger automatically captures line/filename but it is generally quite unhelpful - see https://github.com/ubclaunchpad/rocket2/issues/519).

Using something like `module` gives you a starting point for figuring out where things went wrong, and makes it easier to group problems from related code

Reply:
Thanks Robert for the link, I didn't see that at first!

But yea I agree we could definitely use `module` but I think we should be more specific with our module naming as opposed to just using `express` since that's an external module and not one that we're implementing.

Reply:
```suggestion
const log = parentLogger.child({ module: "router" });
```

is what I would normally call this (messages here should be routing-level logs - shouldn't be too much application logic here)

Reply:
noted, thank you for the feedback and comments!